File: DSRFQ.Web/Modules/Company/Groups/GroupsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Company.Columns;

[ColumnsScript("Company.Groups")]
[BasedOnRow(typeof(GroupsRow), CheckNames = true)]
public class GroupsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }

    [EditLink] public string Name { get; set; }
    public string Description { get; set; }
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}

File: DSRFQ.Web/Modules/ServerTypes/Company/GroupsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { GroupsRow } from "./GroupsRow";

export interface GroupsColumns {
    Id: Column<GroupsRow>;
    Name: Column<GroupsRow>;
    Description: Column<GroupsRow>;
    InsertDate: Column<GroupsRow>;
    InsertBy: Column<GroupsRow>;
    UpdateDate: Column<GroupsRow>;
    UpdateBy: Column<GroupsRow>;
    DeleteDate: Column<GroupsRow>;
    DeleteBy: Column<GroupsRow>;
}

export class GroupsColumns extends ColumnsBase<GroupsRow> {
    static readonly columnsKey = 'Company.Groups';
    static readonly Fields = fieldsProxy<GroupsColumns>();
}